core/src/test/java/org/apache/calcite/test/RelBuilderTest.java

```diff
@@ -3048,4 +3048,30 @@ private void checkExpandTable(RelBuilder builder, Matcher<RelNode> matcher) {
         + "    LogicalTableScan(table=[[scott, DEPT]])\n";
     assertThat(root, hasTree(expected));
   }
+
+  @Test public void testRexSubQueryIn() {
+
+    final RelBuilder builder = RelBuilder.create(config().build());
+    RelNode root = builder.scan("EMP")
+        .filter(
+            builder.in(
+                builder.scan("DEPT")
+                    .filter(
+                        builder.call(SqlStdOperatorTable.EQUALS,
+                            builder.field("DNAME"),
+                            builder.literal("AAA")))
+                    .project(builder.field("DEPTNO"))
+                    .build(),
+                ImmutableList.of(builder.field("DEPTNO"))))
+        .build();
+
```

Comment:
Could you please add a comment about the equivalent sql for better understanding?

Reply:
done